src/main/java/com/sparta/outsourcing/domain/menu/service/MenuService.java

```diff
@@ -50,42 +74,49 @@ public MenuPatchRespDto patchMenu(
         return new MenuPatchRespDto(menuId, dto);
     }
 
+    @Transactional
+    public MenuDeleteRespDto deleteMenu(
+            Long menuId, Long storeId, LoginUser loginUser) {
 
-    public List<MenuGetRespDto> getMenuList(Long storeId) {
-        storeRepository.findById(storeId).orElseThrow(()
-                -> new CustomApiException(ErrorCode.STORE_NOT_FOUND));
-
-        List<Menu> menus = menuRepository.findAllByStoreStoreId(storeId);
+        User user = loginUser.getUser();
+        checkUserAuthority(user);
 
-        return menus.stream().map(MenuGetRespDto::new).toList();
-    }
+        Store store = storeRepository.findById(storeId).orElseThrow(()
+                -> new CustomApiException(ErrorCode.STORE_NOT_FOUND));
 
+        if (user.getEmail().equals(store.getOwner().getEmail())) {
+            throw new CustomApiException(ErrorCode.NO_AUTHORITY);
+        }
 
-    public MenuGetRespDto getMenu(Long menuId, Long storeId) {
+        Menu menu = menuRepository.findById(menuId).orElseThrow(()
+                -> new CustomApiException(ErrorCode.MENU_NOT_FOUND));
 
-        Menu menu = check(menuId, storeId);
-        return new MenuGetRespDto(menu);
-    }
+        if (!Objects.equals(store.getStoreId(), menu.getStore().getStoreId())) {
+            throw new CustomApiException(ErrorCode.STORE_NOT_OWN);
+        }
 
-    @Transactional
-    public MenuDeleteRespDto deleteMenu(Long menuId, Long storeId) {
-        Menu menu = check(menuId, storeId);
-        menu.deleted();
+        menu.deleted(true);
         return new MenuDeleteRespDto(menu);
     }
 
-    private Menu check(Long menuId, Long storeId) {
+    public List<MenuGetRespDto> getMenuList(Long storeId) {
+
         Store store = storeRepository.findById(storeId).orElseThrow(()
                 -> new CustomApiException(ErrorCode.STORE_NOT_FOUND));
 
-        Menu menu = menuRepository.findById(menuId).orElseThrow(()
-                -> new CustomApiException(ErrorCode.MENU_NOT_FOUND));
+        return menuRepository.findAllByStoreStoreId(storeId)
+                .stream()
+                .map(MenuGetRespDto::new)
+                .filter(m -> !m.isDeleted())
+                .toList();
```

Comment:
```suggestion
        return menuRepository.findAllByStoreStoreId(storeId)
                .stream()
                .filter(menu -> !menu.isDeleted())
                .map(MenuGetRespDto::new)
                .toList();
```
filter가 map의 위에 있어야 작동할 것 같습니다.